lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -2030,20 +1926,34 @@ estimateTxSize skeleton =
     --   }
     sizeOf_WitnessSet
         = sizeOf_SmallMap
-        + sizeOf_VKeyWitnesses
-        + sizeOf_NativeScripts txMintOrBurnScripts
-        + maybe 0 (determinePaymentTemplateSize txMintOrBurnScripts) txPaymentTemplate
+        + sizeOf_VKeyWitnesses numberOf_VkeyWitnesses
+        + maybe 0 determinePaymentTemplateSize txPaymentTemplate
+        -- FIXME: Payment template needs to be multiplied with number of inputs
         + sizeOf_BootstrapWitnesses numberOf_BootstrapWitnesses
-      where
-        -- ?0 => [* vkeywitness ]
-        sizeOf_VKeyWitnesses
-            = (if numberOf_VkeyWitnesses > 0
-                then sizeOf_Array + sizeOf_SmallUInt else 0)
-            + sizeOf_VKeyWitness * numberOf_VkeyWitnesses
 
-        -- ?1 => [* native_script ]
+-- ?5 => withdrawals
+sizeOf_Withdrawals :: Integer -> Integer
+sizeOf_Withdrawals n
```

Comment:
Looking at this function (and similar functions) in isolation, I have a few questions:
1. What does the `n` represent? Is it a count of the number of withdrawals?
2. If the value `n` can never be negative, then can we use `Natural` instead?
3. If the result of the function is the size of this part of the transaction, then can we return a value of type `TxSize` (for consistency?)

Reply:
> What does the n represent? Is it a count of the number of withdrawals?

Yes.

> If the value n can never be negative, then can the function accept a value of type Natural instead?

Probably, but I'm trying to not think too much besides keeping the changes small.

> If the result of the function is the size of this part of the transaction, then can we return a value of type TxSize (for consistency?)

Similarly, I'd prefer not to change this now.